panther_lights/src/scheduler_node.py

```diff
@@ -76,6 +83,41 @@ def _scheduler_timer_cb(self, *args) -> None:
             if success:
                 self._led_e_stop_state = self._e_stop_state
 
+        if self._charger_state != self._charger_connected:
+            self._charger_state = self._charger_connected
+            if self._charger_connected:
+                self._charging_battery_timer_cb()  # immediately trigger timers callback
+                self._charging_battery_timer = rospy.Timer(
+                    rospy.Duration(self._charging_battery_anim_period),
+                    self._charging_battery_timer_cb,
+                )
+
+    def _charging_battery_timer_cb(self, *args) -> None:
+        if not self._charger_connected:
+            self._charging_battery_timer.shutdown()
+            self._charging_percentage = -1.0
+            self._led_e_stop_state = None
+            return
```

Comment:
```suggestion
        if not self._charger_connected:
            self._charging_battery_timer.shutdown()
            self._charging_battery_timer.join()
            del self._charging_battery_timer
            self._charging_percentage = -1.0
            self._led_e_stop_state = None
            return
```

Since timers inherit from threads let's better close it properly. Timer implementation doesn't handle this case